backend/db_migrations/1584979214020_add-no-duplicate-players-in-matches.ts

```diff
@@ -0,0 +1,19 @@
+/* eslint-disable @typescript-eslint/camelcase */
+import { MigrationBuilder, ColumnDefinitions } from 'node-pg-migrate'
+
+export const shorthands: ColumnDefinitions | undefined = undefined
+
+export async function up(pgm: MigrationBuilder): Promise<void> {
+  pgm.addConstraint('Matches', 'NO_DUPLICATE_PLAYERS_IN_TEAM_1', {
+    check: '"Team1Player1Id" != "Team1Player2Id"',
+  })
+  pgm.addConstraint('Matches', 'NO_DUPLICATE_PLAYERS_IN_TEAM_2', {
+    check: '"Team2Player1Id" != "Team2Player2Id"',
+  })
+  pgm.addConstraint('Matches', 'NO_DUPLICATE_PLAYERS_IN_MATCHES', {
+    check: '"Team1Player1Id" != "Team2Player1Id" AND ' +
```

Comment:
i'd use backticks ` , so you don't need to concatenate the strings

Reply:
I used this way for easier formatting and at the same time not creating unnecessary whitespaces in the constraint definition. with backsticks and if I keep those on separate lines there will spaces from the beginning of the line if I understand it correctly

Reply:
IMO, you don't have to worry about that - if you add a backslash at the end of the line, it doesn't add a new line. Theoretically you can use tagged templates (https://muffinresearch.co.uk/removing-leading-whitespace-in-es6-template-strings/) to remove the leading whitespaces.

I'm not going to impose the backtick style, you can keep this, but I personally strongly prefer readability over a few added whitespaces (which might be trimmed somewhere down the road)

Reply:
cool, didn't know about that! fixed with common-tags here https://github.com/salsita/foosball-rating/pull/129/commits/1f5c04659da17d5c5fa7e47053799e439a44ca71